pkg/orchestrator/selection/selector/node_selector.go

```diff
@@ -42,45 +42,46 @@ func (n NodeSelector) AllMatchingNodes(ctx context.Context, job *models.Job) ([]
 	return nodeInfos, nil
 }
 func (n NodeSelector) TopMatchingNodes(ctx context.Context, job *models.Job, desiredCount int) ([]models.NodeInfo, error) {
-	filteredNodes, err := n.rankAndFilterNodes(ctx, job)
+	possibleNodes, rejectedNodes, err := n.rankAndFilterNodes(ctx, job)
 	if err != nil {
 		return nil, err
 	}
-	if len(filteredNodes) < desiredCount {
+	if len(possibleNodes) < desiredCount {
```

Comment:
`possibleNodes` contains the same content as `filteredNode` did, right? And were just returning the rejected nodes here for better error messages?

Reply:
Yeah that's it. I renamed it because in my head `filtered` could mean both "filtered in" and "filtered out" and I wanted to make it clearer which was which